test/spec/java_spec.rb

```diff
@@ -91,6 +93,10 @@
           app.deploy do |app|
             expect(app.run('java -version')).to include(java_version)
 
+            EXPECTED_ON_PATH.each do |name|
+              expect(app.run("which #{name}")).not_to eq('')
```

Comment:
If running this command fails with an error (eg due to `heroku run` not working, or say if there was a typo in the command name `which`), it seems like this test might still pass? Hatchet doesn't validate the exit status of the one-off dyno, and combined stdout and stderr afaik?

I wonder if an ~`endswith /<name>` check or even a ~`matches /app/.*/<name>` might be less fragile? (The latter would also ensure you're not matching against some other java etc, not that it's in the base image at the moment, but for completeness)